plugins/modules/system/disk_info.py

```diff
@@ -0,0 +1,167 @@
+        default: all
+        type: str
+    filter:
+        description:
+            - This is the parameter used to filter only the specific information from the output like freespace, mountpoint, fstype, and so on.
+        type: list
+        elements: str
+author:
+    - Saranya Sridharan (@saranyasridharan)
+    - Swetha M (@swetm)
+    - Nagendra Kini (@NagendraKini)
+    - Anup Kumar (@anuppoojari)
+    - Mehmoodkha Pathan (@Mehmoodpathan)
+'''
+
+EXAMPLES = '''
+- name: Get disk info
+  community.general.disk_info:
+
+- name: Get Disk info for Particular disk info
+  community.general.disk_info:
+    name: diskname
+
+- name: Filter the output parameters for specific disk
+  community.general.disk_info:
+    name: diskname
+    filter:
+      - mountpoint
+      - freespace
+      - totalsize
+
+- name: Filter the output parameters for all disk
+  community.general.disk_info:
+    filter:
+      - fstype
+      - usedspace
+      - capacity_percentage
+'''
+
+
+RETURN = '''
+ansible_facts:
+    description: disk information only filtered parameter values if filter option is given.
+    returned: dictionary of disks informations
+    type: dict
+    sample: {
+        '/dev/sda': {
+             'capacity_percentage': '2%'
+        }
+    }
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+import traceback
+PSUTIL_IMP_ERR = None
+try:
+    import psutil
+    HAS_PSUTIL = True
+except ImportError:
+    PSUTIL_IMP_ERR = traceback.format_exc()
+    HAS_PSUTIL = False
+
+
+# To get the disk partitions information
+def disk_info(devicename):
+    output = {}
+    partitions = psutil.disk_partitions()
+    found = False
+    for partition in partitions:
+        if not partition.device.startswith("/dev/"):
+            continue
+        output[partition.device] = {}
+        output[partition.device]['mountpoint'] = partition.mountpoint
+        output[partition.device]['fstype'] = partition.fstype
+        try:
+            partition_usage = psutil.disk_usage(partition.mountpoint)
+        except PermissionError:
+            # this can be catched due to the disk that
+            # isn't ready
+            continue
+        output[partition.device]['totalsize'] = partition_usage.total
+        output[partition.device]['usedspace'] = partition_usage.used
+        output[partition.device]['freespace'] = partition_usage.free
+        output[partition.device]['capacity_percentage'] = partition_usage.percent
+        if partition.device == devicename:
+            found = True
+            break
+    if devicename == "all":
+        return output
+    if devicename in output:
+        return {devicename: output[devicename]}
+    return {}
+
+
+# To filter the required values from the output
+def filter_result(output, filtervalue):
+    filteroutput = {}
+    for key in output:
+        filteroutput[key] = {}
```

Comment:
Why not simply do
```suggestion
        filteroutput[key] = {param: output[param] for param in filter}
```
and remove the `for` loop?

(Assuming you added `choices` as requested.)